SpeechResponder/Personality.cs

```diff
@@ -175,17 +176,19 @@ public Personality Copy(string name, string description)
             string iname = name.ToLowerInvariant();
             string copyPath = Constants.DATA_DIR + @"\personalities\" + iname + ".json";
             // Ensure it doesn't exist
-            if (!File.Exists(copyPath))
+            if (File.Exists(copyPath))
             {
-                ToFile(copyPath);
+                return null; // UI for this is the caller's job
             }
-            // Load the personality back in
+
+            // copy the personality via serializing and deserializing
+            ToFile(copyPath);
             Personality newPersonality = FromFile(copyPath);
```

Comment:
So...
1. We write `this` (the current active personality) to the new file.
2. We immediately read a copy back into memory to create a new object.
3. We adjust the personality metadata (name and description).
4. We write it again.

The current active personality is already in memory (as `this`). Why not skip the file system for steps 1 and 2?

e.g.
```cs
        public Personality Copy(string name, string description)
        {
            // Tidy the name up to avoid bad characters
            string regexSearch = new string(Path.GetInvalidFileNameChars()) + new string(Path.GetInvalidPathChars());
            Regex r = new Regex(string.Format("[{0}]", Regex.Escape(regexSearch)));
            name = r.Replace(name, "");

            string iname = name.ToLowerInvariant();
            string copyPath = Constants.DATA_DIR + @"\personalities\" + iname + ".json";
            // Ensure it doesn't exist
            if (!File.Exists(copyPath))
            {
                // Change its name and description then save the new personality
                Personality newPersonality = this.Copy();
                newPersonality.Name = name;
                newPersonality.Description = description;
                newPersonality.ToFile();
                return newPersonality;
            }
            // The file name already exists so throw an exception. Handling this is the caller's job
            throw new InvalidOperationException();
        }
```

Reply:
Yeah, hitting the file system unnecessarily isn't great.

Reply:
You are right, this is highly inefficient. Will fix in my 2nd take.